test/integration/physics_system.cc

```diff
@@ -2264,3 +2270,207 @@ TEST_F(PhysicsSystemFixtureWithDart6_10,
+  // (here, the model is falling down, so only z should be updated)
+  for (size_t i = 1; i <= nIters; i++)
+  {
+    EXPECT_GT(collisionPoses[i-1].Pos().Z(), collisionPoses[i].Pos().Z()) 
+            << "Model should be falling down.";
+  }
+  
+  // the first part checks that:
+  // - the collision poses are updated correctly
+  // - linear velocities (world/local) are updated.
+  for (size_t i = 1; i < halfIters; i++)
+  {
+    // --- LIN ACC (WORLD, LOCAL)
+    // linear acc is constant = gravity (both world and local)
+    EXPECT_NEAR(linAccs[i].Length(), gravity.Length(), 1e-2);
+    EXPECT_NEAR(worldLinAccs[i].Length(), gravity.Length(), 1e-2);
+    // local should mostly be X , world should be -Z
+    EXPECT_NEAR(linAccs[i].X(), gravity.Length(), 1e-2) << "Local linear acceleration should be along X axis";
+    EXPECT_NEAR(worldLinAccs[i].Z(), -gravity.Length(), 1e-2) << "World Linear acceleration should be along -Z axis";
+    
+    // --- LIN VEL (WORLD, LOCAL)
+    // linear velocity should keep increasing, both local and world
+    EXPECT_LT(linVels[i-1].Length(), linVels[i].Length()) << "Local linear velocity should keep increasing";
+    EXPECT_LT(worldLinVels[i-1].Length(), worldLinVels[i].Length()) << "World linear velocity should keep increasing";
+    // local should mostly be X , world should be -Z
+    EXPECT_GT(linVels[i-1].X(), 0) << "Local linear vel should be positive in x";
+    EXPECT_LT(worldLinVels[i-1].Z(), 0) << "World linear vel should be negative in z";
+
+    // --- ANG ACC (WORLD, LOCAL)
+    // angular acc is constant = 0 (both world and local)
+    EXPECT_NEAR(angAccs[i].Length(), 0, 1e-2);
+    EXPECT_NEAR(worldAngAccs[i].Length(), 0, 1e-2);
+    // --- ANG VEL (WORLD, LOCAL)
+    // angular velocity is constant
+    EXPECT_NEAR(angVels[i].Length(), 0, 1e-2);
+    EXPECT_NEAR(worldAngVels[i].Length(), 0, 1e-2);
+  }
+  
+  // Second part, we applied a varying wrench, and we should see:
+  //  - linear acceleration (world/local) are updated
+  //  - angular acceleration (world/local) are updated
+  //  - angular velocities (world/local) are updated
+  for (size_t i = halfIters + 1; i < nIters; i++)
+  {
+    // --- LIN ACC (WORLD, LOCAL)
+    EXPECT_LT(linAccs[i-1].Length(), linAccs[i].Length()) << "Local Linear Acceleration should be increasing.";
+    EXPECT_LT(worldLinAccs[i-1].Length(), worldLinAccs[i].Length()) << "World Linear Acceleration should be increasing.";
+    
+    // --- ANG ACC (WORLD, LOCAL)
+    EXPECT_LT(angAccs[i-1].Length(), angAccs[i].Length()) << "Local Angular Acceleration should be increasing.";
+    EXPECT_LT(worldAngAccs[i-1].Length(), worldAngAccs[i].Length()) << "World Angular Acceleration should be increasing.";
+
+    // --- ANG VEL (WORLD, LOCAL)
+    EXPECT_LT(angVels[i-1].Length(), angVels[i].Length()) << "Local Angular Velocity should be increasing.";
+    EXPECT_LT(worldAngVels[i-1].Length(), worldAngVels[i].Length()) << "World Angular Velocity should be increasing.";
```

Comment:
Could you add a check that the norm of the `x` and `y` components (`y` and `z` for non-world quantities) of (World)LinearVelocity and (World)LinearAcceleration of are zero before `halfIters` and positive after `halfIters`? This would check that the position offset of the collision is taken into acount properly.

Reply:
Sure ! I wasn't sure how to test that, thanks for the tip.